shop/src/main/java/ro/msg/learning/shop/service/StockService.java

```diff
@@ -0,0 +1,89 @@
+package ro.msg.learning.shop.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.exception.ProductCategoryException;
+import ro.msg.learning.shop.exception.StockException;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.model.Stock;
+import ro.msg.learning.shop.repository.ILocationRepository;
+import ro.msg.learning.shop.repository.IProductRepository;
+import ro.msg.learning.shop.repository.IStockRepository;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class StockService {
+    private static final String ERROR_MESSAGE = "stock not found for the id ";
+    private final IStockRepository stockRepository;
+
+    private final IProductRepository productRepository;
```

Comment:
Usually the repository of an entity should be only used in the service of the respective entity. This would mean that you would have to inject here the services of the other entitites